flow/server/flow_subscriber_endpoint.go

```diff
@@ -18,25 +18,72 @@
 package server
 
 import (
+	"github.com/skydive-project/skydive/common"
 	"github.com/skydive-project/skydive/flow"
+	"github.com/skydive-project/skydive/logging"
 	ws "github.com/skydive-project/skydive/websocket"
 )
 
 // FlowSubscriberEndpoint sends all the flows to its subscribers.
 type FlowSubscriberEndpoint struct {
-	pool ws.StructSpeakerPool
+	common.RWMutex
+	pool         ws.StructSpeakerPool
+	nsSubscriber map[string]ws.Speaker
 }
 
+const flowNS = "flow"
+
 // SendFlows sends flow to the subscribers
 func (fs *FlowSubscriberEndpoint) SendFlows(flows []*flow.Flow) {
-	msg := ws.NewStructMessage("flow", "store", flows)
-	fs.pool.BroadcastMessage(msg)
+	fs.RLock()
+	_, ok := fs.nsSubscriber[flowNS]
+	fs.RUnlock()
+
+	if ok {
+		msg := ws.NewStructMessage(flowNS, "store", flows)
+		fs.pool.BroadcastMessage(msg)
+	}
+}
+
+// OnConnected Server interface
+func (fs *FlowSubscriberEndpoint) OnConnected(c ws.Speaker) {
+	ns := flowNS
+
+	if namespaces, ok := c.GetHeaders()["X-Websocket-Namespace"]; ok {
+		if len(namespaces) > 0 {
+			ns = namespaces[0]
```

Comment:
How about adding all the namespaces here with the same speaker if the subscriber gives more than one NS
``` 
for _, namespace := range namespaces {
fs.nsSubscriber[namespace] = c
}
```
Just a doubt, not sure whether it is useful or not :-)

Reply:
Indeed, going to adapt